src/components/badge/index.js

```diff
@@ -47,6 +47,7 @@ export default class Badge extends BaseComponent {
   }
 
   render() {
+    this.generateStyles();
```

Comment:
I think we should avoid using generateStyles in render (and in general) since it creates the whole StyleSheet object and it might be heavy to do in each render.
 
It would be better to move the functionality based on the label prop to other functions that returns the badge style parts.  similar to what we do in the Button component (getContainerSizeStyle, getOutlineStyle, etc..)
This way we can also add unit tests for this functionalities. 

Generally I would like to slowly avoid using the generateStyle and deprecate it eventually.